src/lib/mina_base/zkapp_account.ml

```diff
@@ -154,6 +156,20 @@ module Stable = struct
   end
 end]
 
+type t =
+  ( Zkapp_state.Value.t
+  , (Side_loaded_verification_key.t, F.t) With_hash.t option
+  , Mina_numbers.Zkapp_version.t
+  , F.t
+  , Mina_numbers.Global_slot.t
+  , bool )
+  Poly.t
+[@@deriving sexp, equal, compare, hash, yojson]
+
+let () =
+  let _f : unit -> (t, Stable.Latest.t) Type_equal.t = fun () -> Type_equal.T in
```

Comment:
It's code to enforce that types `t` and `Stable.Latest.t` are the same. 

The same technique is used in several places where we have `no_toplevel_latest_type` (I think Matt came up with that).

It would be nicer to add `constraint t = Stable.Latest.t`, but that generates compiler warnings.

Reply:
Oh I see. I though we did `type t = Stable.Latest.t = {...` for that to work. is this a different way of doing it?

Reply:
That works for records and sums, there's an error about "differing kinds" if you try that here, for a type with parameters.

Reply:
That works for records and sums, there's an error about "differing kinds" if you try that here, for a type with parameters.